src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -688,7 +689,12 @@ internal static TimeSpan CalculateTimeoutTime(DateTime startTime, DateTime curre
             var isSolutionLoadRestore = _isFirstRestore &&
                 request.RestoreSource == RestoreOperationSource.Implicit;
             _isFirstRestore = false;
+            string timeSinceLastRestoreCompletedTime = isSolutionLoadRestore ?
+                "0" :
+                (DateTimeOffset.UtcNow - _lastRestoreCompletedTime).TotalSeconds.ToString();
```

Comment:
If first restore operation is not `implicit` (not sure if it's possible scenario) then `isSolutionLoadRestore` is false and same time we have risk of `_lastRestoreCompletedTime` not set yet. Did we account that here?

Reply:
If there are no CPS projects, then there'd be no solution load restore. 
I think the logic is still correct though, as we really only care about the differences between the different restores.

Reply:
I misunderstood your comment. Fixed.